app/controllers/conference_controller.rb

```diff
@@ -15,6 +15,9 @@ def schedule
     @rooms = @conference.venue.rooms if @conference.venue
     @events = @conference.program.events
     @dates = @conference.start_date..@conference.end_date
+    confirmed_events = @conference.program.events.where(state: 'confirmed')
+    @scheduled_events = confirmed_events.where('start_time IS NOT NULL').order(:start_time)
```

Comment:
Maybe it's worth to introduce also a scope for this.
(Which you can also use in the next line combined with a ``not``)